classes.py

```diff
@@ -0,0 +1,264 @@
+
+    def transform(self,x):
+
+        for col in x.columns:
+            
+            wdays=x[col].dt.dayofweek
+            month=x[col].dt.month
+            day=x[col].dt.day
+
+            x[col+'_'+'week_sin']=np.sin(2*np.pi*wdays/self.week_freq)
+            x[col+'_'+'week_cos']=np.cos(2*np.pi*wdays/self.week_freq)
+
+            x[col+'_'+'month_sin']=np.sin(2*np.pi*month/self.month_freq)
+            x[col+'_'+'month_cos']=np.cos(2*np.pi*month/self.month_freq)
+
+            x[col+'_'+'month_day_sin']=np.sin(2*np.pi*day/self.month_day_freq)
+            x[col+'_'+'month_day_cos']=np.cos(2*np.pi*day/self.month_day_freq)
+
+            del x[col]
+
+        return x
+
+    def get_feature_names(self):
+
+        self.feature_names
+
+
+
+class VarSelector(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,feature_names):
+        
+        
+        self.feature_names=feature_names
+        
+    
+    def fit(self,x,y=None):
+
+        return self 
+    
+    def transform(self,x) :
+        
+        return x[self.feature_names]
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+
+class DataFrameImputer(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        
+        self.feature_names=[]
+        self.impute_dict={}
+        
+    
+    def fit(self,x,y=None):
+        
+        self.feature_names=x.columns
+        
+        for col in x.columns:
+            
+            if x[col].dtype=='O':
+                self.impute_dict[col]='missing'
+            else :
+                self.impute_dict[col]=x[col].median()
+                
+        return self
+    
+    def transform(self,x) :
+        
+        return x.fillna(self.impute_dict)
+        
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+    
+class string_clean(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,replace_it,replace_with):
+        
+        
+        self.feature_names=[]
+        self.replace_it=replace_it
+        self.replace_with=replace_with        
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+
+        return self 
+    
+    def transform(self,x) :
+        
+        for col in x.columns:
+
+            x[col]=x[col].str.replace(self.replace_it,self.replace_with)
+        
+        return x
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+    
+    
+
+class count_vectorizer(BaseEstimator,TransformerMixin):
+
+    def __init__(self):
+        
+        
+        self.feature_names=[]
+        
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+        
+        return self
+    
+    def transform(self,x) :
+        
+        for col in x.columns:
+            x[col]=x[col].TfidfVectorizer(analyzer='word',stop_words='english',max_df=0.8,min_df=0.01,max_features=200)
+            x[col]=x[col].toarray()
```

Comment:
Are you sure tfidf vectoriser works like this . why is there no fit and transform here ?